src/core/tsi/ssl_transport_security.cc

```diff
@@ -1538,15 +1539,40 @@ static tsi_result ssl_handshaker_next(tsi_handshaker* self,
   // If there are received bytes, process them first.
   tsi_ssl_handshaker* impl = reinterpret_cast<tsi_ssl_handshaker*>(self);
   tsi_result status = TSI_OK;
-  size_t bytes_consumed = received_bytes_size;
   size_t bytes_written = 0;
   if (received_bytes_size > 0) {
-    status = ssl_handshaker_process_bytes_from_peer(impl, received_bytes,
-                                                    &bytes_consumed, error);
-    while (status == TSI_DRAIN_BUFFER) {
-      status = ssl_handshaker_write_output_buffer(self, &bytes_written, error);
-      if (status != TSI_OK) return status;
-      status = ssl_handshaker_do_handshake(impl, error);
+    unsigned char* remaining_bytes_to_write_to_network_io =
```

Comment:
The names of these two variables are confusing.  If I'm understanding the code correctly (but please correct me if I'm not), these aren't the bytes being written to the network; they're the bytes from the peer that we're reading from the network.

Reply:
Agreed that the wording is a bit confusing. Let me clarify what's happening first: we're getting bytes from the peer, and writing these bytes to OpenSSL via the `BIO` object, and the name of this `BIO` object variable is `network_io`.

I'm happy to send a follow-up PR to rename the `BIO` object variable (because it causes issues like this throughout this file), but I'd prefer to keep that change out-of-scope for this PR if it's OK.

In the meantime, I've changed the variable name to something else to avoid confusion.